hypershift-operator/controlplaneoperator-overrides/overrides.go

```diff
@@ -0,0 +1,84 @@
+package controlplaneoperatoroverrides
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+
+	"github.com/openshift/hypershift/support/config"
+	"gopkg.in/yaml.v2"
+)
+
+type CPOOverrides struct {
+	Overrides []CPOOverride `yaml:"overrides,omitempty"`
+}
+
+type CPOOverride struct {
+	Version  string                  `yaml:"version"`
+	CPOImage string                  `yaml:"cpoImage"`
+	Testing  CPOOverrideTestReleases `yaml:"testing"`
+}
+
+type CPOOverrideTestReleases struct {
+	Latest   string `yaml:"latest"`
+	Previous string `yaml:"previous"`
+}
+
+//go:embed assets/overrides.yaml
+var overridesYAML []byte
+
+var (
+	overrides          = mustLoadOverrides()
+	overridesByVersion map[string]*CPOOverride
+)
+
+func init() {
+	initOverridesByVersion()
+}
+
+func IsOverridesEnabled() bool {
+	return os.Getenv(config.CPOOverridesEnvVar) == "1"
+}
+
+func CPOImage(version string) string {
+	return lookupCPOImage(version)
+}
+
+func LatestOverrideTestReleases() (string, string) {
+	if len(overrides.Overrides) > 0 {
+		testingReleases := overrides.Overrides[len(overrides.Overrides)-1].Testing
+		return testingReleases.Latest, testingReleases.Previous
+	}
+	return "", ""
+}
+
+func lookupCPOImage(version string) string {
+	fmt.Println("overridesByVersion", overridesByVersion)
+	if override, exists := overridesByVersion[version]; exists {
```

Comment:
IIRC this was for managed services, they will move to supporting disconnected env soon (?) how should we deal with that in here? Should we put the destination internal registry they will use for that? if that's not the case I think we will need to use a releaseProvider to handle the overrides of the overrides (😅)

Reply:
Like any other image we deliver to service delivery, they will need a ICSP in the mgmt cluster to replace with their local registry.